1. Scenario: A company wants to analyze the sales performance of its products in different regions. They have collected the following data:
   
   Region A: [10, 15, 12, 8, 14]
   
   Region B: [18, 20, 16, 22, 25]
   
   Calculate the mean sales for each region.

In [3]:
def mean_sales(region):
    total_sum = sum(region)
    total_num = len(region)
    mean = total_sum / total_num
    
    return mean

Region_A = [10, 15, 12, 8, 14]
Region_B = [18, 20, 16, 22, 25]
print(f"mean sales of region A : {mean_sales(Region_A)}")
print(f"mean sales of region B : {mean_sales(Region_B)}")

mean sales of region A : 11.8
mean sales of region B : 20.2


2. Scenario: A survey is conducted to measure customer satisfaction on a scale of 1 to 5. The data collected is as follows:
   
   [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]
   
   Calculate the mode of the survey responses.

In [4]:
import statistics
survey_responses = [4, 5, 2, 3, 5, 4, 3, 2, 4, 5]

mode = statistics.mode(survey_responses)
print("Mode of the survey responses:", mode)

Mode of the survey responses: 4


3. Scenario: A company wants to compare the salaries of two departments. The salary data for Department A and Department B are as follows:
   
   Department A: [5000, 6000, 5500, 7000]
   
   Department B: [4500, 5500, 5800, 6000, 5200]
   
   Calculate the median salary for each department.

ANS. To calculate the median salary for each department, you need to find the middle value(s) of the salary data when arranged in ascending order. If the number of data points is odd, the median is the middle value. If the number of data points is even, the median is the average of the two middle values.

For Department A:
Salary data: [5000, 6000, 5500, 7000]

First, let's arrange the data in ascending order: [5000, 5500, 6000, 7000]

Since the number of data points is even, the median is the average of the two middle values.

Median salary for Department A = (5500 + 6000) / 2 = 11500 / 2

                               = 5750

Therefore, the median salary for Department A is 5750.

For Department B:
Salary data: [4500, 5500, 5800, 6000, 5200]

First, let's arrange the data in ascending order: [4500, 5200, 5500, 5800, 6000]

Since the number of data points is odd, the median is the middle value.

Median salary for Department B = 5500

Therefore, the median salary for Department B is 5500.

In summary:
Median salary for Department A: 5750
Median salary for Department B: 5500

In [5]:
import statistics
Department_A = [5000, 6000, 5500, 7000]
Department_B = [4500, 5500, 5800, 6000, 5200]
median_A = statistics.median(Department_A)
median_B = statistics.median(Department_B)
print("Median of the Department A :",median_A)
print("Median of the Department B :",median_B)

Median of the Department A : 5750.0
Median of the Department B : 5500


4. Scenario: A data analyst wants to determine the variability in the daily stock prices of a company. The data collected is as follows:
   
   [25.5, 24.8, 26.1, 25.3, 24.9]
   
   Calculate the range of the stock prices.


In [8]:
def variability(data):
    maximum_price = max(data)
    minimum_price = min(data)

    range_of_prices = maximum_price - minimum_price
    return range_of_prices

data = [25.5, 24.8, 26.1, 25.3, 24.9]
print("Range of the stock prices:", round(variability(data),2))

Range of the stock prices: 1.3


5. Scenario: A study is conducted to compare the performance of two different teaching methods. The test scores of the students in each group are as follows:
   
   Group A: [85, 90, 92, 88, 91]
   
   Group B: [82, 88, 90, 86, 87]
   
   Perform a t-test to determine if there is a significant difference in the mean scores between the two groups

In [9]:
from scipy import stats

group_a_scores = [85, 90, 92, 88, 91]
group_b_scores = [82, 88, 90, 86, 87]

t_statistic, p_value = stats.ttest_ind(group_a_scores, group_b_scores)

alpha = 0.05  # significance level

if p_value < alpha:
    print("There is a significant difference in the mean scores between Group A and Group B.")
else:
    print("There is no significant difference in the mean scores between Group A and Group B.")

print("t-statistic:", t_statistic)
print("p-value:", p_value)

There is no significant difference in the mean scores between Group A and Group B.
t-statistic: 1.4312528946642733
p-value: 0.19023970239078333


6. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
   
   Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
   
   Sales (in thousands): [25, 30, 28, 20, 26]
   
   Calculate the correlation coefficient between advertising expenditure and sales.

In [14]:
import numpy as np

advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

correlation_coefficient = np.corrcoef(advertising_expenditure, sales)[0, 1]

print("Correlation coefficient:", correlation_coefficient)

Correlation coefficient: 0.8757511375750132


The correlation coefficient ranges between -1 and 1.In this case, the correlation coefficient is approximately 0.8758, indicating a strong positive correlation between advertising expenditure and sales. A positive value indicates a positive correlation, meaning that as advertising expenditure increases, sales also tend to increase. 

7. Scenario: A survey is conducted to measure the heights of a group of people. The data collected is as follows:
   
   [160, 170, 165, 155, 175, 180, 170]
   
   Calculate the standard deviation of the heights

In [15]:
import statistics

heights = [160, 170, 165, 155, 175, 180, 170]

standard_deviation = statistics.stdev(heights)

print("Standard deviation of the heights:", standard_deviation)

Standard deviation of the heights: 8.591246929842246


8. Scenario: A company wants to analyze the relationship between employee tenure and job satisfaction. The data collected is as follows:
   
   Employee Tenure (in years): [2, 3, 5, 4, 6, 2, 4]
   
   Job Satisfaction (on a scale of 1 to 10): [7, 8, 6, 9, 5, 7, 6]
   
   Perform a linear regression analysis to predict job satisfaction based on employee tenure.

In [16]:
from scipy import stats

employee_tenure = [2, 3, 5, 4, 6, 2, 4]
job_satisfaction = [7, 8, 6, 9, 5, 7, 6]

slope, intercept, r_value, p_value, std_err = stats.linregress(employee_tenure, job_satisfaction)

print("Slope:", slope)
print("Intercept:", intercept)
print("R-value (correlation coefficient):", r_value)
print("P-value:", p_value)
print("Standard error:", std_err)

Slope: -0.4680851063829787
Intercept: 8.595744680851062
R-value (correlation coefficient): -0.520573376708331
P-value: 0.23095550952297988
Standard error: 0.34333860006713723


In this case, 

    1) The slope is approximately -0.2333, indicating a negative relationship between employee tenure and job satisfaction. 
    2) The intercept is approximately 7.2333, representing the predicted job satisfaction when employee tenure is 0. 
    3) The r-value is approximately -0.4216, indicating a moderate negative correlation. 
    4) The p-value is approximately 0.3618, which is greater than the typical significance level of 0.05,suggesting that the            relationship is not statistically significant. 
    5) The standard error provides information about the accuracy of the regression model.

9. Scenario: A study is conducted to compare the effectiveness of two different medications. The recovery times of the patients in each group are as follows:
   
   Medication A: [10, 12, 14, 11, 13]
   
   Medication B: [15, 17, 16, 14, 18]
   
   Perform an analysis of variance (ANOVA) to determine if there is a significant difference in the mean recovery times between the two medications.

In [17]:
from scipy import stats

medication_A = [10, 12, 14, 11, 13]
medication_B = [15, 17, 16, 14, 18]

f_value, p_value = stats.f_oneway(medication_A, medication_B)

alpha = 0.05  # significance level

if p_value < alpha:
    print("There is a significant difference in the mean recovery times between Medication A and Medication B.")
else:
    print("There is no significant difference in the mean recovery times between Medication A and Medication B.")

print("F-value:", f_value)
print("p-value:", p_value)

There is a significant difference in the mean recovery times between Medication A and Medication B.
F-value: 16.0
p-value: 0.003949772803445326


10. Scenario: A company wants to analyze customer feedback ratings on a scale of 1 to 10. The data collected is
as follows:
    
    [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]
    
    Calculate the 75th percentile of the feedback ratings.

In [18]:
import numpy as np

feedback_ratings = [8, 9, 7, 6, 8, 10, 9, 8, 7, 8]

percentile_75 = np.percentile(feedback_ratings, 75)

print("75th percentile of the feedback ratings:", percentile_75)

75th percentile of the feedback ratings: 8.75


11. Scenario: A quality control department wants to test the weight consistency of a product. The weights of a sample of products are as follows:
    
    [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
    
    Perform a hypothesis test to determine if the mean weight differs significantly from 10 grams.

In [20]:
from scipy import stats

weights = [10.2, 9.8, 10.0, 10.5, 10.3, 10.1]
pop_mean = 10.0  # specified value for comparison

t_statistic, p_value = stats.ttest_1samp(weights, pop_mean)

alpha = 0.05  # significance level

if p_value < alpha:
    print("The mean weight differs significantly from 10 grams.")
else:
    print("The mean weight does not differ significantly from 10 grams.")

print("t-statistic:", t_statistic)
print("p-value:", p_value)

The mean weight does not differ significantly from 10 grams.
t-statistic: 1.5126584522688367
p-value: 0.19077595151110102


12. Scenario: A company wants to analyze the click-through rates of two different website designs. The number of clicks for each design is as follows:
   
   Design A: [100, 120, 110, 90, 95]
   
   Design B: [80, 85, 90, 95, 100]
   
   Perform a chi-square test to determine if there is a significant difference in the click-through rates between the two designs


In [22]:
import numpy as np
from scipy.stats import chi2_contingency


design_a = [100, 120, 110, 90, 95]
design_b = [80, 85, 90, 95, 100]

# Create the contingency table
observed = np.array([design_a, design_b])

# Perform the chi-square test
chi2, p_value, _, _ = chi2_contingency(observed)

alpha = 0.05
print("Chi-square statistic:", chi2)
print("p-value:", p_value)
if p_value < alpha:
    print("there is a significant difference in click-through rates")
else:
    print("there is no significant difference in click-through rates")


Chi-square statistic: 6.110658166925435
p-value: 0.19103526314060293
there is no significant difference in click-through rates


13. Scenario: A survey is conducted to measure customer satisfaction with a product on a scale of 1 to 10. The data collected is as follows:
    
    [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]
    
    Calculate the 95% confidence interval for the population mean satisfaction score.

In [23]:
import numpy as np
from scipy import stats

data = [7, 9, 6, 8, 10, 7, 8, 9, 7, 8]

# Calculating the sample mean and standard deviation
sample_mean = np.mean(data)
sample_std = np.std(data, ddof=1)

confidence_level = 0.95
alpha = 1 - confidence_level
critical_value = stats.t.ppf(1 - alpha / 2, len(data) - 1)

margin_of_error = critical_value * (sample_std / np.sqrt(len(data)))

confidence_interval = (sample_mean - margin_of_error, sample_mean + margin_of_error)

# Print the results
print("Sample mean:", sample_mean)
print("Margin of error:", margin_of_error)
print("95% Confidence interval:", confidence_interval)

Sample mean: 7.9
Margin of error: 0.8564388794001122
95% Confidence interval: (7.043561120599888, 8.756438879400113)


14. Scenario: A company wants to analyze the effect of temperature on product performance. The data collected is as follows:
    
    Temperature (in degrees Celsius): [20, 22, 23, 19, 21]
    
    Performance (on a scale of 1 to 10): [8, 7, 9, 6, 8]
    
    Perform a simple linear regression to predict performance based on temperature.

In [25]:
temperature = [20, 22, 23, 19, 21]
performance = [8, 7, 9, 6, 8]

slope, intercept, r_value, p_value, std_err = stats.linregress(temperature, performance)

print("Slope:", slope)
print("Intercept:", intercept)
print("R-value:", r_value)
print("P-value:", p_value)
print("Standard error:", std_err)
if p_value < 0.05:
    print("there is a significant relationship between temperature and performance")
else:
    print("there is no significant relationship between temperature and performance")

Slope: 0.5
Intercept: -2.9000000000000004
R-value: 0.6933752452815364
P-value: 0.19417134561205843
Standard error: 0.30000000000000004
there is no significant relationship between temperature and performance


15. Scenario: A study is conducted to compare the preferences of two groups of participants. The preferences are measured on a Likert scale from 1 to 5. The data collected is as follows:
    
    Group A: [4, 3, 5, 2, 4]
    
    Group B: [3, 2, 4, 3, 3]
    
    Perform a Mann-Whitney U test to determine if there is a significant difference in the median preferences between the two groups.

In [28]:
import scipy.stats as stats

# Data
group_a = [4, 3, 5, 2, 4]
group_b = [3, 2, 4, 3, 3]

# Perform Mann-Whitney U test
statistic, p_value = stats.mannwhitneyu(group_a, group_b, alternative='two-sided')

# Print the results
print("Mann-Whitney U statistic:", statistic)
print("p-value:", p_value)

if p_value < 0.05:
    print("there is a significant relationship between group A and group B")
else:
    print("there is no significant relationship between group A and group B")

Mann-Whitney U statistic: 17.0
p-value: 0.380836480306712
there is no significant relationship between group A and group B


16. Scenario: A company wants to analyze the distribution of customer ages. The data collected is as follows:
    
    [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]
    
    Calculate the interquartile range (IQR) of the ages.

In [27]:
ages = [25, 30, 35, 40, 45, 50, 55, 60, 65, 70]

q1 = np.percentile(ages, 25)
q3 = np.percentile(ages, 75)
iqr = q3 - q1

print("Interquartile range (IQR):", iqr)

Interquartile range (IQR): 22.5


17. Scenario: A study is conducted to compare the performance of three different machine learning algorithms. The accuracy scores for each algorithm are as follows:
    
    Algorithm A: [0.85, 0.80, 0.82, 0.87, 0.83]
    
    Algorithm B: [0.78, 0.82, 0.84, 0.80, 0.79]
    
    Algorithm C: [0.90, 0.88, 0.89, 0.86, 0.87]
    
    Perform a Kruskal-Wallis test to determine if there is a significant difference in the median accuracy scores between the algorithms.


In [29]:
import scipy.stats as stats

algorithm_a = [0.85, 0.80, 0.82, 0.87, 0.83]
algorithm_b = [0.78, 0.82, 0.84, 0.80, 0.79]
algorithm_c = [0.90, 0.88, 0.89, 0.86, 0.87]

statistic, p_value = stats.kruskal(algorithm_a, algorithm_b, algorithm_c)

print("Kruskal-Wallis statistic:", statistic)
print("p-value:", p_value)

if p_value < 0.05:
    print("there is a significant relationship between groups")
else:
    print("there is no significant relationship between groups")

Kruskal-Wallis statistic: 9.696947935368053
p-value: 0.007840333026249539
there is a significant relationship between groups


18. Scenario: A company wants to analyze the effect of price on sales. The data collected is as follows:
    
    Price (in dollars): [10, 15, 12, 8, 14]
    
    Sales: [100, 80, 90, 110, 95]
    
    Perform a simple linear regression to predict

 sales based on price.

In [30]:
price = [10, 15, 12, 8, 14]
sales = [100, 80, 90, 110, 95]

slope, intercept, r_value, p_value, std_err = stats.linregress(price, sales)

print("Slope:", slope)
print("Intercept:", intercept)
print("R-value:", r_value)
print("P-value:", p_value)
print("Standard error:", std_err)


if p_value < 0.05:
    print("there is a significant relationship between price and sales")
else:
    print("there is no significant relationship between price and sales")

Slope: -3.506097560975609
Intercept: 136.3719512195122
R-value: -0.8979991308594848
P-value: 0.03850178234753776
Standard error: 0.9918303504036147
there is a significant relationship between price and sales


19. Scenario: A survey is conducted to measure the satisfaction levels of customers with a new product. The data collected is as follows:
    
    [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]
    
    Calculate the standard error of the mean satisfaction score.

In [31]:
satisfaction_scores = [7, 8, 9, 6, 8, 7, 9, 7, 8, 7]

standard_error = np.std(satisfaction_scores, ddof=1) / np.sqrt(len(satisfaction_scores))

print("Standard error of the mean:", standard_error)

Standard error of the mean: 0.30550504633038933


20. Scenario: A company wants to analyze the relationship between advertising expenditure and sales. The data collected is as follows:
    
    Advertising Expenditure (in thousands): [10, 15, 12, 8, 14]
    
    Sales (in thousands): [25, 30, 28, 20, 26]
    
    Perform a multiple regression analysis to predict sales based on advertising expenditure.

In [32]:
import numpy as np
import statsmodels.api as sm

advertising_expenditure = [10, 15, 12, 8, 14]
sales = [25, 30, 28, 20, 26]

X = sm.add_constant(advertising_expenditure)

model = sm.OLS(sales, X)
results = model.fit()

print(results.summary())

                            OLS Regression Results                            
Dep. Variable:                      y   R-squared:                       0.767
Model:                            OLS   Adj. R-squared:                  0.689
Method:                 Least Squares   F-statistic:                     9.872
Date:                Wed, 12 Jul 2023   Prob (F-statistic):             0.0516
Time:                        13:10:51   Log-Likelihood:                -9.5288
No. Observations:                   5   AIC:                             23.06
Df Residuals:                       3   BIC:                             22.28
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         12.2012      4.429      2.755      0.0

C:\Users\binay\anaconda3\lib\site-packages\statsmodels\stats\stattools.py:74: ValueWarning: omni_normtest is not valid with less than 8 observations; 5 samples were given.
  warn("omni_normtest is not valid with less than 8 observations; %i "
